waku/node/waku_node.nim

```diff
@@ -1213,9 +1213,34 @@ proc lightpushPublish*(
       error "lightpush publish error", error = msg
       return lighpushErrorResult(INTERNAL_SERVER_ERROR, msg)
   
-  # TODO: check eligibilityProof here
+  # Checking eligibility proof of Lightpush request
+  debug "in lightpushPublish"
+  debug "eligibilityProof: ", eligibilityProof
+  if node.peerManager.eligibilityManager.isNone():
+    # the service node doesn't want to check eligibility
+    debug "eligibilityManager is disabled - skipping eligibility check"
+  else:
+    debug "eligibilityManager is enabled"
+    var em = node.peerManager.eligibilityManager.get()
+    # FIXME: where should I init eligibilityManager?
+    # FIXME: how to reuse EthClient from CLI arguments?
+    debug "initializing eligibilityManager..."
+    let ethClient = "https://sepolia.infura.io/v3/470c2e9a16f24057aee6660081729fb9"
+    em = await EligibilityManager.init(ethClient)
+    debug "checking eligibilityProof..."
+    let txNonExistent = TxHash.fromHex("0x0000000000000000000000000000000000000000000000000000000000000000")
+    let expectedToAddress = Address.fromHex("0xe8284Af9A5F3b0CD1334DBFaf512F09BeDA805a3")
+    let expectedValueWei = 30000000000000.u256
+    
+    let isEligible = await em.isEligibleTxId(
+      eligibilityProof.get(), expectedToAddress, expectedValueWei
+    )
+    if isEligible.isErr():
+      let msg = "Eligibility check failed!"
+      #debug msg
+      return lighpushErrorResult(PAYMENT_REQUIRED, msg)
 
-  # TODO: eligibilityProof must be already checked if present
+  debug "Eligibility check passed!"
```

Comment:
`node_factory.nim` will call some proc here with the config items as parameters.